templates/shared_services/sonatype-nexus-vm/scripts/deploy_nexus_container.sh

```diff
@@ -20,7 +20,18 @@ while true; do
   ((docker_pull_timeout--));
 done
 
+# Deduce memory available to Java. Either 3/4 of the system RAM, or a set minimum
+# shellcheck disable=SC2002
+mem_total_mb=$(( $(cat /proc/meminfo | head -1 | awk '{ print $2 }') / 1024 ))
+java_mem=2703
+if [ $mem_total_mb -gt 4096 ]; then
+  java_mem=$(( mem_total_mb * 3 / 4 ))
```

Comment:
Is there a reason we use 3/4, rather than total, maybe minus 2GB for the OS?

Reply:
```suggestion
  java_mem=$(( mem_total_mb - 2048 ))
```

Reply:
I just thought that as the machine gets bigger it may be advisable to leave more for the host OS, I don't like running machines at the limit, it may be detrimental to overall performance.

E.g., I don't know if Nexus does any in-memory caching of files. If not, it may help performance to leave some memory for the OS, so the OS filesystem cache can help.

I don't feel strongly about it, so I don't mind which way we go with this commit.

Reply:
Let's leave it as you suggested, I doubt most will need more than 8GB so would work out the same anyway.